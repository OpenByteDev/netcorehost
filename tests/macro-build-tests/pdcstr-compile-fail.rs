fn main() {
    // with internal nul 
    let _ = netcorehost::pdcstr!("\0");
    let _ = netcorehost::pdcstr!("somerandomteststring\0");
    let _ = netcorehost::pdcstr!("somerandomteststring\0somerandomteststring");
    let _ = netcorehost::pdcstr!("somerandomteststring\0somerandomteststring");
}
