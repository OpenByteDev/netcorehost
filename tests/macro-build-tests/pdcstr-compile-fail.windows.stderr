error[E0080]: evaluation of constant value failed
 --> tests/macro-build-tests/pdcstr-compile-fail.rs:3:13
  |
3 |     let _ = netcorehost::pdcstr!("\0");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'invalid NUL value found in string literal', $DIR/tests/macro-build-tests/pdcstr-compile-fail.rs:3:13
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `netcorehost::pdcstr` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: evaluation of constant value failed
 --> tests/macro-build-tests/pdcstr-compile-fail.rs:4:13
  |
4 |     let _ = netcorehost::pdcstr!("somerandomteststring\0");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'invalid NUL value found in string literal', $DIR/tests/macro-build-tests/pdcstr-compile-fail.rs:4:13
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `netcorehost::pdcstr` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: evaluation of constant value failed
 --> tests/macro-build-tests/pdcstr-compile-fail.rs:5:13
  |
5 |     let _ = netcorehost::pdcstr!("somerandomteststring\0somerandomteststring");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'invalid NUL value found in string literal', $DIR/tests/macro-build-tests/pdcstr-compile-fail.rs:5:13
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `netcorehost::pdcstr` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
 --> tests/macro-build-tests/pdcstr-compile-fail.rs:5:13
  |
5 |     let _ = netcorehost::pdcstr!("somerandomteststring\0somerandomteststring");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this note originates in the macro `$crate::pdcstring::windows::widestring::u16cstr` which comes from the expansion of the macro `netcorehost::pdcstr` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
 --> tests/macro-build-tests/pdcstr-compile-fail.rs:4:13
  |
4 |     let _ = netcorehost::pdcstr!("somerandomteststring\0");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this note originates in the macro `$crate::pdcstring::windows::widestring::u16cstr` which comes from the expansion of the macro `netcorehost::pdcstr` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
 --> tests/macro-build-tests/pdcstr-compile-fail.rs:3:13
  |
3 |     let _ = netcorehost::pdcstr!("\0");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this note originates in the macro `$crate::pdcstring::windows::widestring::u16cstr` which comes from the expansion of the macro `netcorehost::pdcstr` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
 --> tests/macro-build-tests/pdcstr-compile-fail.rs:3:13
  |
3 |     let _ = netcorehost::pdcstr!("\0");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this note originates in the macro `$crate::pdcstring::windows::widestring::u16cstr` which comes from the expansion of the macro `netcorehost::pdcstr` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
 --> tests/macro-build-tests/pdcstr-compile-fail.rs:4:13
  |
4 |     let _ = netcorehost::pdcstr!("somerandomteststring\0");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this note originates in the macro `$crate::pdcstring::windows::widestring::u16cstr` which comes from the expansion of the macro `netcorehost::pdcstr` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
 --> tests/macro-build-tests/pdcstr-compile-fail.rs:5:13
  |
5 |     let _ = netcorehost::pdcstr!("somerandomteststring\0somerandomteststring");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this note originates in the macro `$crate::pdcstring::windows::widestring::u16cstr` which comes from the expansion of the macro `netcorehost::pdcstr` (in Nightly builds, run with -Z macro-backtrace for more info)
